sparktk-core/src/main/scala/org/trustedanalytics/sparktk/models/timeseries/arimax/ArimaxModel.scala

```diff
@@ -0,0 +1,224 @@
+    validateFormatVersion(formatVersion, validFormatVersions: _*)
+    val m: ArimaxModelTkMetaData = SaveLoad.extractFromJValue[ArimaxModelTkMetaData](tkMetadata)
+
+    // Define the spark-ts ARIMAX model
+    val sparkModel = new SparkTsArimaxModel(m.p, m.d, m.q, m.xregMaxLag, m.coefficients, m.includeOriginalXreg, m.includeIntercept)
+
+    // Create ArimaxModel to return
+    ArimaxModel(m.timeseriesColumn, m.xColumns, m.p, m.d, m.q, m.xregMaxLag, m.includeOriginalXreg, m.includeIntercept, m.initParams, sparkModel)
+  }
+}
+
+case class ArimaxModel private[arimax] (timeseriesColumn: String,
+                                        xColumns: Seq[String],
+                                        p: Int,
+                                        d: Int,
+                                        q: Int,
+                                        xregMaxLag: Int,
+                                        includeOriginalXreg: Boolean = true,
+                                        includeIntercept: Boolean = true,
+                                        initParams: Option[Seq[Double]],
+                                        arimaxModel: SparkTsArimaxModel) extends Serializable {
+
+  /**
+   * An intercept term
+   */
+  def c: Double = arimaxModel.coefficients(0)
+
+  /**
+   * Coefficient values: AR with increasing degrees
+   */
+  def ar: Seq[Double] = arimaxModel.coefficients.slice(1, p + 1)
+
+  /**
+   * Coefficient values: MA with increasing degrees
+   */
+  def ma: Seq[Double] = arimaxModel.coefficients.slice(p + 1, p + q + 1)
+
+  /**
+   * Coefficient values: xreg with increasing degrees
+   */
+  def xreg: Seq[Double] = arimaxModel.coefficients.drop(p + q + 1)
+
+  def predict(frame: Frame,
+              predictTimeseriesColumn: String,
+              predictXColumns: Seq[String]): Frame = {
+    require(frame != null, "Frame cannot be null.")
+    require(predictXColumns.length == xColumns.length, "Number of columns for train and predict should be the same")
+
+    // Get vector or y values and matrix of x values
+    val predictFrameRdd = new FrameRdd(frame.schema, frame.rdd)
+    val (yVector, xMatrix) = TimeSeriesFunctions.getYAndXFromFrame(predictFrameRdd, predictTimeseriesColumn, predictXColumns)
+
+    // Find the maximum lag and fill that many spots with nulls, followed by the predicted y values
+    val maxLag = max(arimaxModel.p, arimaxModel.q)
+    val ts = new BreezeDenseVector(yVector.toArray)
+
+    val predictions = Array.fill(maxLag) { null } ++ arimaxModel.predict(ts, xMatrix).toArray
+    val numPredictions = predictions.length
+
+    val dataWithPredictions = frame.rdd.zipWithIndex().map {
+      case (row: Row, index: Long) =>
+        if (numPredictions > index)
+          Row.fromSeq(row.toSeq :+ predictions(index.toInt))
+        else
+          Row.fromSeq(row.toSeq :+ null)
+    }
+
+    val schemaWithPredictions = frame.schema.addColumn(Column("predicted_y", DataTypes.float64))
```

Comment:
Why predicted_y ? It should be sth like this (scala) s"predicted_$predictedTimeseriesColumn"